client/components/jetpack/intro-pricing-banner/index.tsx

```diff
@@ -25,14 +27,34 @@ const IntroPricingBanner: React.FC = () => {
 	const isSmallScreen = useBreakpoint( '<660px' );
 
 	const windowBoundaryOffset = useMemo( () => {
-		if ( isJetpackCloud() || isConnectStore() ) {
+		if ( isJetpackCloud() ) {
 			return CLOUD_MASTERBAR_HEIGHT;
+		} else if ( isConnectStore() ) {
+			return CONNECT_STORE_HEIGHT;
 		}
-
 		return CALYPSO_MASTERBAR_HEIGHT;
 	}, [] );
+
 	const [ barRef, hasCrossed ] = useDetectWindowBoundary( windowBoundaryOffset );
 
+	const prevHasCrossed = usePrevious( hasCrossed );
+
+	useEffect( () => {
+		if ( ! shouldShowCart ) {
+			return;
+		}
+		const navHeaderEle = document.getElementsByClassName( 'header__content-background-wrapper' );
+		if ( ! navHeaderEle || ! navHeaderEle.length ) {
+			return;
+		}
+
+		if ( hasCrossed ) {
+			navHeaderEle[ 0 ].classList.remove( 'header__content-background-wrapper--sticky' );
```

Comment:
This way we would have very touch coupling between elements, that actually used separately across the codebase. If we would show banner on some other page, or change the classname, this will break the behaviour. We need to think of a way that classnames would be passed or something.

Reply:
Hi Andrii, yeah, i agree with you on this. I tried to figure out a better, less coupled way to do it but couldn't find a quick solution. So, I decided to do this way to ship it quickly and also considering `<IntroPricingBanner/>` is rendered only on the pricing page, and it shouldn't impact other pages.

Reply:
And also, to me the `<IntroPricingBanner/>` should be refactored better to just render the banner and other logic like attaching the cart and making ti sticky should be handled by the parent component rendering it, since the pricing page might need the sticky banner but not the other pages rendering it.